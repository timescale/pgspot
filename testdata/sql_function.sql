
-- unsafe call inside function
CREATE FUNCTION sqlfunc() RETURNS TEXT LANGUAGE SQL AS $$ SELECT unsafe_call3(); $$;

-- safe call inside function
CREATE FUNCTION sqlfunc() RETURNS TEXT LANGUAGE SQL AS $$ SELECT safe_call6(); $$ SET search_path TO 'pg_catalog';

-- unsafe call inside function, since search_path for body is not safe
SET search_path TO pg_catalog;
CREATE FUNCTION sqlfunc() RETURNS TEXT LANGUAGE SQL AS $$ SELECT unsafe_call10(); $$;

CREATE FUNCTION sqlfunc() RETURNS TEXT LANGUAGE SQL AS $$
SELECT unsafe_call13();
SET search_path TO pg_catalog;
SELECT safe_call15();
RESET search_path;
SELECT unsafe_call17();
SELECT foo.safe_call18();
$$;
