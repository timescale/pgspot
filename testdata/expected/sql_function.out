PS005: Function without explicit search_path: sqlfunc() at line 2
PS016: Unqualified function call: unsafe_call3 at line 2
PS005: Function without explicit search_path: sqlfunc() at line 10
PS016: Unqualified function call: unsafe_call10 at line 10
PS005: Function without explicit search_path: sqlfunc() at line 11
PS016: Unqualified function call: unsafe_call13 at line 11
PS016: Unqualified function call: unsafe_call17 at line 11

 Errors: 0 Warnings: 7 Unknown: 0 

